### PR TITLE
SCRUM-102: refactor docker and storage stuff

### PR DESCRIPTION
**Pull request by filkovsp**
- refactor docker compose so that it could be used locally and ran in Docker without setting up environment variables. `.env.local` file still can be used to replace defaults.
- refactor naming convention for Storage classes.
- refactor naming for `containers`, `volume`, `docker compose group`

  image

### CODE REVIEW

**Review by Ne4upara — APPROVED** [review]
It`s ok.

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
     restart: always
     environment:
       POSTGRES_PASSWORD: postgres
     ports:
       - "5432:5432"
     volumes:
-      - pg_marketplace:/var/lib/postgresql/data
-      - ../src/main/sql/schema.sql:/docker-entrypoint-initdb.d/schema.sql
+      - khutir-db-data:/var/lib/postgresql/data
+      - ./dbsetup.sql:/docker-entrypoint-initdb.d/dbsetup.sql
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply docker/docker-compose.yml:29]: yes indeed.
we needed, coz we have to create our own db,
but unfortunately we can't do that with composer. 
to get what we need we've got 2 alternatives: 
1. doing it with Dockerfile and create db as bart of docker build process
2. attach db init script the way we do it here now - which is the most simple one.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -1,37 +1,43 @@
...
 
   mail:
-    container_name: mailhog-server
-    image: mailhog/mailhog:v1.0.1
+    container_name: "khutir-mail"
+    image: mailhog/mailhog:latest
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply docker/docker-compose.yml:33]: oh, good suggestion, 
do we need to change it permanently here ? 
if yes, could you create a jira ticket for that ?

File: docker/README.md [hunk 3]
```diff
@@ -20,9 +20,9 @@ This project is a Spring Boot application with PostgreSQL as the database. Docke
 To start the application and the PostgreSQL database using Docker Compose, run:
 
 ```bash
-docker-compose --file ./docker/docker-compose.yml up -d
+docker compose --file ./docker/docker-compose.yml up -d
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply docker/README.md:23]: coz "docker-compose" is legacy now and not supported any more.
https://docs.docker.com/compose/releases/migrate

Docker documentation suggests using "docker compose" instead.
https://docs.docker.com/compose/intro/compose-application-model

File: src/main/java/com/khutircraftubackend/storage/CloudinaryService.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
 import java.util.Map;
 
 @RequiredArgsConstructor
-public class CloudinaryServiceImpl implements StorageService {
+public class CloudinaryService implements StorageService {
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply src/main/java/com/khutircraftubackend/storage/CloudinaryService.java:13]: this is legacy approach. as well as adding "i" as prefix to an interface name.
if we sick to all of these legacy methodologies, we would probably name classes this way:
```java
public interface ICar {
  ... 
}

public class CarTruckmpl implements ICar {
  ... 
}
```

nowadays, we do it better:
```java
public interface Animal {
  ... 
}

public class Elephant implements Animal {
  ...
}
```

File: src/main/java/com/khutircraftubackend/storage/config/CloudinaryConfig.java [hunk 5]
```diff
@@ -5,8 +5,10 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
 
 @Configuration
+@Profile("!local")
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply src/main/java/com/khutircraftubackend/storage/config/CloudinaryConfig.java:11]: no, it's actually not :) 
but yes, it's not very good to have `@Profile("!local")` here either 
a better annotation would probably be `@Profile("cloudinary")` 
but for us `!local` right now is ok.
reason why it's needed - for running in local dev env, without Environment Variables for Cloudinary being setup. 
it becomes a problem if me or our QA or someone from frontend devs try to run backend with local profile
CloudinaryConfig still tries to create this bean, hence it requires to have all the security params to be provided for it, even for local profile.

hence, 
ideally, it should annotated with `@Profile("cloudinary")`  and only involved into spring context when it's needed.

File: src/main/resources/application-local.yml [hunk 6]
```diff
@@ -2,18 +2,27 @@ spring:
...
+    driver-class-name: org.postgresql.Driver
+    url: ${SPRING_DATASOURCE_URL:jdbc:postgresql://localhost:5432/khutir_craft}
+    username: ${SPRING_DATASOURCE_USERNAME:postgres}
+    password: ${SPRING_DATASOURCE_PASSWORD:postgres}
 
   jpa:
     generate-ddl: true
     hibernate:
-      ddl-auto: create
+      # ACHTUNG!!! keep `ddl-auto: update` for local profile.
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply src/main/resources/application-local.yml:14]: if we put here "create" - it'll work the same way as flyway - drops and creates all tables.
try:
1. mvn clean 
2. make sure you have some users or any other data in other tables
3. ddl-auto value from `update` with `create`
4. start backend
5. check data - not, it'll be wiped out

so, no point then to favour using ddl-auto in dev mode, if it wipes out data.
better probably to rely on flyway in this case.

File: src/main/resources/application-local.yml [hunk 7]
```diff
@@ -24,6 +33,10 @@ spring:
           starttls:
             enable: false
 
+jwt:
+  secret: ${JWT_SECRET:dummy-secret-phrase}
```
[the remark this answers is not in the document]
Comment (filkovsp) [reply src/main/resources/application-local.yml:37]: please note, 
only application-local has fall back values for all the Environment Variables. 
this is done on purpose here. To make it possible to use local profile and run backend by QA and FrontEnd people without configuring everything. this is acceptable for local dev configuration only. 
at the same time we still have possibility to override the fall back "dummy-secret-phrase" for "JWT_SECRET" from `.env.local` file if we want to.

File: docker/docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (YP0n) [reply docker/docker-compose.yml:33]: Most likely it is necessary.
Ticket jira for that? Yes.

File: src/main/java/com/khutircraftubackend/storage/CloudinaryService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (YP0n) [reply src/main/java/com/khutircraftubackend/storage/CloudinaryService.java:13]: There is an opinion that by looking at the class name, we should understand what it does.

**Pull request merged**